crates/tx/src/types.rs

```diff
@@ -1039,6 +1084,27 @@ impl Tx {
         }
     }
 
+    /// Add new default commitments to the transaction. Returns false if the
+    /// commitment is already contained in the set
+    #[cfg(any(test, feature = "testing"))]
+    pub fn push_default_inner_tx(&mut self) -> bool {
+        self.header.batch.insert(TxCommitments::default())
+    }
+
+    /// Add a new inner tx to the transaction. Returns `false` if the
+    /// commitments already existed in the collection. This function expects a
+    /// transaction carrying a single inner tx as input
```

Comment:
Is there a reason this function is limited to a single inner tx?

Reply:
At the moment the support to build batched transactions is very limited. We'll probably want to rework the way we construct txs so for now I've implemented something minimal. The caller can invoke this function multiple times if needed